dbunit/src/main/java/org/jboss/arquillian/persistence/dbunit/dataset/scriptable/ScriptableTable.java

```diff
@@ -0,0 +1,121 @@
+ */
+public class ScriptableTable implements ITable {
+
+    //any non digit char followed by ':' followed by 1 or more chars e.g: js: new Date().toString()
+    private final Pattern scriptEnginePattern = Pattern.compile(".*\\D.*:.+");
+
+    private static Logger log = Logger.getLogger(ScriptableTable.class.getName());
+
+    private Map<String, ScriptEngine> engines;
+
+    private ScriptEngineManager manager;
+
+    private ITable delegate;
+
+    private ScriptableTable() {
+        throw new UnsupportedOperationException("This constructor should not be called");
+    }
+
+    public ScriptableTable(ITable delegate) {
+        this.delegate = delegate;
+        engines = new HashMap<String, ScriptEngine>();
+        manager = new ScriptEngineManager();
+    }
+
+    @Override
+    public ITableMetaData getTableMetaData() {
+        return delegate.getTableMetaData();
+    }
+
+    @Override
+    public int getRowCount() {
+        return delegate.getRowCount();
+    }
+
+    @Override
+    public Object getValue(int row, String column) throws DataSetException {
+        Object value = delegate.getValue(row, column);
+        if (value != null && scriptEnginePattern.matcher(value.toString()).matches()) {
+            ScriptEngine engine = getScriptEngine(value.toString().trim());
+            if (engine != null) {
+                Object scriptResult = getScriptResult(value.toString(), engine);
+                if (scriptResult != null) {
+                    value = scriptResult;
+                } else {
+                    throw new RuntimeException(String.format("Could not evaluate script expression for table '%s', column '%s'.", getTableMetaData().getTableName(), column));
+                }
+            }
+        }
+        return value;
+    }
+
+    /**
+     * Parses table cell to get script engine
+     *
+     * @param value the table cell
+     * @return scriptEngine
+     */
+    private ScriptEngine getScriptEngine(String value) {
+        String engineName = value.substring(0, value.indexOf(":"));
+        if (engines.containsKey(engineName)) {
+            return engines.get(engineName);
+        } else {
+            ScriptEngine engine = manager.getEngineByName(engineName);
+            if (engine != null) {
+                engines.put(engineName, engine);
+            } else {
+                log.warning(String.format("Could not find script engine with name %s in classpath", engineName));
+            }
+            return engine;
+        }
+
+    }
+
+    /**
+     * Evaluates the script expression
+     *
+     * @return script expression result or null if any evaluation error
+     */
+    private Object getScriptResult(String script, ScriptEngine engine) {
+        String scriptToExecute = script.substring(script.indexOf(":") + 1);
+        try {
+            return engine.eval(scriptToExecute);
+        } catch (Exception e) {
+            return null;
```

Comment:
That's probably not what we want, right? Maybe some logging at least? Why shouldn't we fail in such a case?